src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/KestrelThread.cs

```diff
@@ -85,6 +95,8 @@ public void AllowStop()
 
         public void Stop(TimeSpan timeout)
         {
+            DisposeConnections(timeout);
```

Comment:
I would delete `KestrelThread.AllowStop()` (nothing calls it) now that this needs to be called for gracefull shutdown, and this requires the thread to be running.